docs/advanced/developer-guide.md

```diff
@@ -0,0 +1,293 @@
+---
+title: "Developer Guide"
+layout: default
+sort: 1
+---
+
+# Developer Guide
+{: .no_toc }
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+---
+
+## Building from Source
+
+**Download the source code:**
+
+```
+git clone https://github.com/kubernetes-sigs/node-feature-discovery
+cd node-feature-discovery
+```
+
+### Docker Build
+
+
+**Build the container image:**<br>
+See [customizing the build](#customizing-the-build) below for altering the
+container image registry, for example.
+
+```
+make
+```
+
+**Push the container image:**<br>
+Optional, this example with Docker.
+
+```
+docker push <IMAGE_TAG>
+```
+
+**Change the job spec to use your custom image (optional):**
+
+To use your published image from the step above instead of the
+`k8s.gcr.io/nfd/node-feature-discovery` image, edit `image`
+attribute in the spec template(s) to the new location
+(`<registry-name>/<image-name>[:<version>]`).
+
+
+### Deployment
+
+The `yamls` makefile generates deployment specs matching your locally built
+image. See [build customization](#customizing-the-build) below for configurability,
+e.g. changing the deployment namespace.
+```
```

Comment:
Is a good Markdown lint to leave a space from text to scriplets 
Also to enhance text highlighting I like to use "```bash" os shell when declaring scriplets on MD

Reply:
Fixed